src/main/java/com/rarchives/ripme/App.java

```diff
@@ -44,6 +42,33 @@
     public static String stringToAppendToFoldername = null;
     private static final History HISTORY = new History();
 
+    public static class IPCThread implements Runnable {
+        public void run() {
+            String dataFromProcess;
+            String capitalizedSentence;
+            try {
+                ServerSocket welcomeSocket = new ServerSocket(6789);
+                while (true) {
+                    Socket connectionSocket = welcomeSocket.accept();
+                    BufferedReader inFromClient = new BufferedReader(new InputStreamReader(connectionSocket.getInputStream()));
+
+                    DataOutputStream outToClient = new DataOutputStream(connectionSocket.getOutputStream());
+                    dataFromProcess = inFromClient.readLine();
+                    System.out.println("Received: " + dataFromProcess);
+                    capitalizedSentence = dataFromProcess.toUpperCase() + 'n';
+                    outToClient.writeBytes(capitalizedSentence);
+                    rip(new URL(dataFromProcess));
+                    welcomeSocket.close();
+                }
+            } catch(IOException e) {
+                e.printStackTrace();
+            } catch (Exception e) {
+                e.printStackTrace();
```

Comment:
Can't you combine both catches into 1 ?

Reply:
> Can't you combine both catches into 1 ?

Yea this PR is a bit rough right now, I'll fix that once i have some more free time to work on this